### ISSUE TITLE
No longer putting opened windows into a single window

### ISSUE THREAD

**Issue by GuardianMajor** [VanCoding/One-Window#8]
It had been working perfectly for a long time but suddenly today out of the blue, it has stopped working and everything just opens up in a new window and does not get wrangled back into a single window as tabs anymore. I don't know if a recent code update to Chromium broke it or what, but it would be great if we can figure out what went wrong. If you need anything let me know. Thank you.

**Comment by VanCoding** [comment]
Oh darn, there indeed seems to be a problem in the latest version.
Thanks for reporting this!

I'm working on it ;)

**Comment by VanCoding** [comment]
I've pulished a fix now. I hope it solved the problems you are encountering!
Please let me know if it does not.

Thank you!

**Comment by GuardianMajor** [comment]
I am happy to report, its a simple but effective extension. I will give it a test and get back to you.

**Comment by GuardianMajor** [comment]
Unfortunately my friend, it is still pooched. It is now wrangling the windows into a single window but not properly as it did before. Before if you have two windows open and the new windows were initiated in say window 2 it would create the tabs in there but now it will throw all of of them in the first window that was created no matter where it originates which is not previous normal behavior and seriously makes it flawed still.

**Comment by VanCoding** [comment]
Okay, thanks. I'll further look into that issue ;)

**Comment by VanCoding** [comment]
Ok, I could now reproduce the issue myself and am now looking for a solution.

**Comment by VanCoding** [comment]
I've published a path that should fix the issue to the webstore. Can you verify the fix?

**Comment by GuardianMajor** [comment]
I will take a look now, give me a few minutes and I will get back to you on it.

**Comment by GuardianMajor** [comment]
Ok, we are getting there but a few things that are still different. It is now respecting WHICH window the others were opened and grabs them and puts them in THAT active window, which is a good fix, thank you and confirmed its working.

However, before, it would TAB the windows based on the order they were opened which was great but unfortunately now, it will tabs them in reverse order (meaning last window opened, is tabbed as the first tab) and all next to the initiating tab, which is new behavior.

Additionally, it is not grabbing all windows that are opened. If they windows are opened using a script (like I do most of the time) they are being grabbed, be it in the wrong order) but some windows that pop up using another method (such as some social media share buttons, or even some websites that use popups for dialogs) will not get grabbed where they were in the past.

Let me know if i can be any more help.

**Comment by VanCoding** [comment]
Wow, thanks for the close checkup!

The order will be easy to fix, but in order to fix the other issue with some windows not getting merged, it would be helpful to get some examples of where/what windows don't get merged.

**Comment by GuardianMajor** [comment]
The quickest link I can find is this: http://www.playbuzz.com/christopherwallace10/which-historical-couple-are-you?ff=1 just click on the share facebook and you'll see what I mean. If I get anymore I will make note of it and let you know.

**Comment by linmo** [comment]
Hi good day - since this week we have the same issue, pop-ups no longer open up in tabs.
In our corporate environment we still need to use Chrome v36 as the latest Chrome updates broke 2 of our applications that run in Chrome.
Can you please make your superb extensions backwards compatible?

**Comment by VanCoding** [comment]
@GuardianMajor Interesting, the tab that gets created when clicking the share on facebook button on the site you mentioned gets merged for me in verion 3.0.4. Does it always happen, or only in certain circumstances?

Can you maybe tell me your Chrome version & Operating system?
I'm on Windows 7 x64 and Chrome 40.0.2214.115 m.

**Comment by VanCoding** [comment]
Well, after testing a bit longer, it suddenly stopped working here as well and only worked again after restarting the extension. I'm further looking into it.

Thank you for you patience!

**Comment by linmo** [comment]
Hi @VanCoding are you looking into the possible version issue too?

**Comment by VanCoding** [comment]
@linmo I don't think there's a version issue ;) It doesn't work on all versions. It will work again when I figure out how to fix this bug.

**Comment by VanCoding** [comment]
I have now published another fix. I really hope that it works as expected now.
If it works, I'll start working on a fix for the tab order.

**Comment by linmo** [comment]
thanks @VanCoding - this now seems to be working for us again!

**Comment by GuardianMajor** [comment]
@VanCoding 
It happens all the time now and even on the Bing website too. I am on Windows 8.1 Pro x64 and Chromium Version 36.1.1.3 and Chrome Version 41.0.2272.76 m.
As the other user also mentioned, on the 36 it will not tab but on the 41 it will. Does that help?

**Comment by GuardianMajor** [comment]
Patrik it seems to be working on the windows pop ups, good job. I will keep and eye on it to catch anything if it should pop up again. Thank you for the fix. Let me know when you have the tab order one resolved and I will test that out too.

**Comment by GuardianMajor** [comment]
Ok upon further testing, while 3.0.4 did temporarily fix the ones that were not being captured before, it broke the existing ones that were working fine and opened through scripts such as window.open

Once I saw the 3.0.5 update I figured I further test to see if the glitch is resolved and found that while for the most part if grabs them, there is a HUGE delay in performance that wasn't there before. Not sure if the additional code or whatnot but it has become so slow in grabbing the windows that most of them become fully loaded and painted before they are grabbed and a few are just left there without being grabbed at all.

Not to mention they are still being forced added to the far left and and not as they open being placed on the far right like they should and have been in the past. Let me know if you want me to test anything else.
